mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java

```diff
@@ -80,4 +105,11 @@ void start() {
         return eventSerializer.toJson(event).getBytes(StandardCharsets.UTF_8);
     }
 
+    private OutboundMessage generateMessage(RoutingKeyConverter.RoutingKey routingKey, byte[] payload) {
+        Map<String, Object> headers = ImmutableMap.of(EVENT_BUS_ID, eventBusId.getId().toString());
```

Comment:
Can we have a `EventBusId::asString` for doing this and not leaking the UID representatiopn?